src/main/java/ch/bfh/ti/i4mi/mag/mhd/iti65/Iti65RequestConverter.java

```diff
@@ -1000,20 +1000,30 @@ public Telecom transform(ContactPoint contactPoint) {
     }
 
 	/**
-	 * FHIR Organization -> XDS Organization
+	 * FHIR Organization -> IPF Organization (XDS XON)
 	 * @param org
 	 * @return
 	 */
 	public org.openehealth.ipf.commons.ihe.xds.core.metadata.Organization transform(Organization org) {
-		org.openehealth.ipf.commons.ihe.xds.core.metadata.Organization result = new org.openehealth.ipf.commons.ihe.xds.core.metadata.Organization();
+		final var result = new org.openehealth.ipf.commons.ihe.xds.core.metadata.Organization();
     	result.setOrganizationName(org.getName());
-    	Identifier identifier = org.getIdentifierFirstRep();
+    	final Identifier identifier = org.getIdentifierFirstRep();
     	if (identifier != null) {
-    		result.setIdNumber(identifier.getValue());
-    		result.setAssigningAuthority(new AssigningAuthority(noPrefix(identifier.getSystem())));
-    	}
-    	return result;
-    }
+			if (identifier.hasSystem()) {
```

Comment:
proposal: change line 1011 to `if ((identifier != null) && identifier.hasValue()) {`, change line 1021 to simply `else {`.

Reply:
Done!